### PR TITLE
Skip wait_boot for autoyast installations

### PR DESCRIPTION
**Pull request by frankenmichl**
When we deploy the system with autoyast, there is no reboot after the installation and wait_boot never sees a grub menu, so let's skip this in this case.

This will fix https://openqa.suse.de/tests/11998046 and similar

### CODE REVIEW

File: tests/autoyast/console.pm [hunk 1]
```diff
@@ -14,7 +14,7 @@ use Utils::Backends;
 
 sub run {
     my ($self) = @_;
-    $self->wait_boot if is_ipmi;
+    $self->wait_boot if is_ipmi && !get_vat('AUTOYAST');
```
Comment (mdoucha) [inline tests/autoyast/console.pm:17]: Typo.

**Comment by mdoucha** [comment]
Verification run: https://openqa.suse.de/tests/12048649

File: tests/autoyast/console.pm [hunk 2]
```diff
@@ -14,7 +14,13 @@ use Utils::Backends;
 
 sub run {
     my ($self) = @_;
-    $self->wait_boot if is_ipmi;
+    # IPXE boot does not provide boot menu so set pxe_boot_done equals 1 without checking needles
+    my $pxe_boot_done;
+    $pxe_boot_done = 1 if (check_var('IPXE', '1') || check_var('IPXE_UEFI', '1'));
```
Comment (Martchus) [inline tests/autoyast/console.pm:19]:
```suggestion
    my $pxe_boot_done = check_var('IPXE', '1') || check_var('IPXE_UEFI', '1');
```

File: tests/autoyast/console.pm [hunk 3]
```diff
@@ -14,7 +14,13 @@ use Utils::Backends;
...
+
+    # If we didn't see pxe, the reboot is going now
+    $self->wait_boot if is_ipmi and not get_var('VIRT_AUTOTEST') and not $pxe_boot_done;
```
Comment (Martchus) [inline tests/autoyast/console.pm:22]: Normally the symbolic operators are preferred (unless you do command chaining):
```suggestion
    $self->wait_boot if is_ipmi && !get_var('VIRT_AUTOTEST') && !$pxe_boot_done;
```

File: tests/autoyast/console.pm (hunk 2 above)
Comment (frankenmichl) [reply tests/autoyast/console.pm:19]: Good suggestion, thanks!

**Review by mdoucha — CHANGES_REQUESTED** [review]

File: tests/autoyast/console.pm [hunk 4]
```diff
@@ -14,7 +14,12 @@ use Utils::Backends;
 
 sub run {
     my ($self) = @_;
-    $self->wait_boot if is_ipmi;
+    # IPXE boot does not provide boot menu so set pxe_boot_done equals 1 without checking needles
+    my $pxe_boot_done = check_var('IPXE', '1') || check_var('IPXE_UEFI', '1');
+
+    # If we didn't see pxe, the reboot is going now
+    $self->wait_boot if is_ipmi && !get_var('VIRT_AUTOTEST') && !$pxe_boot_done;
```
[the remark this answers is not in the document]
Comment (mdoucha) [reply tests/autoyast/console.pm:21]: For non-IPMI jobs, there should be no `wait_boot` call in either module.

[the remark this answers is not in the document]
Comment (mdoucha) [reply tests/autoyast/console.pm:21]: Hmm, no, never mind. On second thought, this is probably correct as is.

**Comment by mdoucha** [comment]
@Julie-CAO Please have a look whether this will work correctly with VIRT_AUTOTEST.

**Comment by Julie-CAO** [comment]
> @Julie-CAO Please have a look whether this will work correctly with VIRT_AUTOTEST.

thanks for asking. `tests/autoyast/console.pm`  is not called by virt tests, so VIRT_AUTOTEST is not required here.

**Comment by frankenmichl** [comment]
So we can keep this a bit simpler

**Review by Martchus — APPROVED** [review]

**Review by mdoucha — APPROVED** [review]
The first comment is redundant but otherwise LGTM.

**Comment by Martchus** [comment]
Yes, the commits should be squashed. You can do this directly on GitHub when merging.

**Comment by frankenmichl** [comment]
New verification run: https://openqa.suse.de/tests/12091394

**Pull request merged**